File: leetcode/242_valid_anagram/main.go
package main

import "fmt"


func main()  {
	fmt.Println(validAnagram("cat", "rat"))	
	fmt.Println(validAnagram("anagram", "nagaram"))	
	fmt.Println(validAnagram("aacc", "ccac"))	
}

func validAnagram(s, t string) bool {
	hashMap := make(map[rune]int)

	for _, v := range s {
		hashMap[v]++
	}

	for _, v := range t {
		hashMap[v]--
	}

	for _, v := range hashMap {
		if v != 0 {
			return false
		}
	}
	
	return true
}

File: leetcode/242_valid_anagram/main.py
def valid_anagram(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False

    s_count, t_count = {}, {}

    for i in range(len(s)):
        s_count[s[i]] = s_count.get(s[i], 0) + 1
        t_count[t[i]] = t_count.get(t[i], 0) + 1

    for c in s_count:
        if s_count[c] != t_count.get(c, 0):
            return False

    return True    

print(valid_anagram("anagram", "nagaram"))    
print(valid_anagram("cat", "rat"))    
print(valid_anagram("aacc", "ccac"))    